libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -339,4 +354,10 @@ private void startNavigation(DirectionsRoute route) {
       navigation.startNavigation(route);
     }
   }
+
+  public interface Callback {
+    void userOffRoute(OffRouteEvent offRouteEvent);
```

Comment:
Are `userOffRoute` and `fasterRouteFound` always implemented together? 👀 [Interface_segregation_principle](https://en.wikipedia.org/wiki/Interface_segregation_principle) from `SOLID` principles.